### PR TITLE
Remove an impossible possibility during a putIfAbsent

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

**Comment by AbfrmBlr** [comment]
needs rebase

**Comment by henri-tremblay** [comment]
Rebased

**Comment by chrisdennis** [comment]
Abhilash is doing work that will likely pull the loader-writer logic out in to a store layer.  This is going obviously going to be disruptive of lots of this code. It would seem to make more sense to me to hold off on doing anything here until Abhilash was finished.

**Comment by henri-tremblay** [comment]
No. This is in sync with what @AbfrmBlr is doing. The PR comes from a question he asked.

**Comment by AbfrmBlr** [comment]
Yes, but it would be good if we can hold this for a bit.

File: core/src/main/java/org/ehcache/core/EhcacheBase.java [hunk 1]
```diff
@@ -297,9 +297,6 @@ public V putIfAbsent(final K key, final V value) {
         if (put[0]) {
           putIfAbsentObserver.end(PutIfAbsentOutcome.PUT);
           return null;
-        } else if (inCache == null) {
```
Comment (AbfrmBlr) [inline core/src/main/java/org/ehcache/core/EhcacheBase.java]: Other than the fact that this part of code was unnecessary, I was more interested in `Consumer<Boolean> put` to go, which is unlikely and we need it. Thus I still need to make `Store` spi change respectively. So lets wait for the Store layer refactoring.

Comment (henri-tremblay) [reply core/src/main/java/org/ehcache/core/EhcacheBase.java]: Oh! Wait a second. We actually might now that this if was removed. Silly me. Let me look.

Comment (henri-tremblay) [reply core/src/main/java/org/ehcache/core/EhcacheBase.java]: Done. No consumer anymore

Comment (AbfrmBlr) [reply core/src/main/java/org/ehcache/core/EhcacheBase.java]: this helps

**Comment by AbfrmBlr** [comment]
This has test failures

**Comment by chrisdennis** [comment]
@AbfrmBlr since this interacts with your stuff I'll let you review/approve/merge this.

**Comment by henri-tremblay** [comment]
to trigger a build

**Pull request merged**